arrow/obsidian.c

```diff
@@ -0,0 +1,279 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "obsidian.h"
+
+// Initial length of event vectors
+static int INITIAL_LENGTH = 4000;
```

Comment:
If this becomes a performance bottleneck (which I highly doubt), I can probably come up with a heuristic for choosing a good size, based on the number of expected events for the simulated interval.